BUILD.md

```diff
@@ -0,0 +1,127 @@
+# Compiling and installing sparta
```

Comment:
Perhaps we should move this file into the cmake folder, since it only pertains to CMake and doesn't include instructions on the other build system. Or rename it to something like BUILD_CMAKE.md.